src/cpp/fastdds/domain/DomainParticipantFactory.cpp

```diff
@@ -332,6 +332,7 @@ ReturnCode_t DomainParticipantFactory::load_profiles()
         XMLProfileManager::loadDefaultXMLFile();
         default_xml_profiles_loaded = true;
 
+        // Only change default participant qos when not explicitly set by the user
```

Comment:
And only do it the first time. After that, if the default changes any further, no action is taken

Reply:
Already done. See `if (false == default_xml_profiles_loaded)` above

Reply:
Added just before the previous line of code.